GoRogue/Lines.cs

```diff
@@ -274,5 +274,13 @@ private static IEnumerable<Point> Ortho(int startX, int startY, int endX, int en
                 yield return new Point(workX, workY);
             }
         }
+
+        #region IEnumerable<Point> Extensions
+
+        public static int LeftAt(this IEnumerable<Point> self, int y) => self.Where(c => c.Y == y).OrderBy(c => c.X).First().X;
+        public static int RightAt(this IEnumerable<Point> self, int y) => self.Where(c => c.Y == y).OrderBy(c => c.X).Last().X;
+        public static int TopAt(this IEnumerable<Point> self, int x) => self.Where(c => c.X == x).OrderBy(c => c.Y).First().Y;
```

Comment:
This at `BottomAt` may have an issue with `Direction.YIncreasesUpward.  Should be a simple fix to check the value of `YIncreasesUpward` and change whether you return first or last :D